lib/internal/http2/core.js

```diff
@@ -1595,8 +1597,10 @@ class ServerHttp2Stream extends Http2Stream {
       const statusCode = headers[HTTP2_HEADER_STATUS] |= 0;
       if (statusCode === HTTP_STATUS_SWITCHING_PROTOCOLS)
         throw new errors.Error('ERR_HTTP2_STATUS_101');
-      if (statusCode < 100 || statusCode >= 200)
-        throw new errors.RangeError('ERR_HTTP2_INVALID_INFO_STATUS');
+      if (statusCode < 100 || statusCode >= 200) {
+        throw new errors.RangeError('ERR_HTTP2_INVALID_INFO_STATUS',
+                                    statusCode);
```

Comment:
may want to use the pre-transformed `headers[HTTP2_HEADER_STATUS]` value here instead of `statusCode`